test/unit/modules/transport.js

```diff
@@ -308,58 +325,55 @@ describe('transport', function () {
 				describe('for every signature in signatures', function () {
 
 					it('should call __private.receiveSignature with signature', function (done) {
-
 						__private.receiveSignatures({
-							signatures: ['SIGNATURE123', 'SIGNATURE456']
+							signatures: [SAMPLE_SIGNATURE_1, SAMPLE_SIGNATURE_2]
 						}, function (err) {
 							expect(library.schema.validate.called).to.be.true;
 							expect(__private.receiveSignature.calledTwice).to.be.true;
-							expect(__private.receiveSignature.calledWith('SIGNATURE123')).to.be.true;
-							expect(__private.receiveSignature.calledWith('SIGNATURE456')).to.be.true;
+							expect(__private.receiveSignature.calledWith(SAMPLE_SIGNATURE_1)).to.be.true;
+							expect(__private.receiveSignature.calledWith(SAMPLE_SIGNATURE_2)).to.be.true;
 							done();
 						});
 					});
 
 					describe('when __private.receiveSignature fails', function () {
 
-						it('should call library.logger.debug with err and signature', function (done) {
+						beforeEach(function (done) {
 							var err = 'Error processing signature: Error message';
-							__private.receiveSignature = sinon.stub().callsArgWith(1, err); // TODO: Also move to beforeEach
-							library.logger.debug = sinon.spy();
+							__private.receiveSignature = sinonSandbox.stub().callsArgWith(1, err);
+							done();
+						});
 
+						it('should call library.logger.debug with err and signature', function (done) {
 							__private.receiveSignatures({
-								signatures: ['SIGNATURE123', 'SIGNATURE456']
+								signatures: [SAMPLE_SIGNATURE_1, SAMPLE_SIGNATURE_2]
 							}, function (err) {
 								expect(library.schema.validate.called).to.be.true;
 								// If any of the __private.receiveSignature calls fail, the whole
```

Comment:
@MaciejBaj Should we create a new issue for this (both `receiveTransactions` and `receiveSignatures`)?
Also, should we:

1. Process transactions/signatures in series but in case of failure, skip that one and continue the processing the others.
2. Process all transactions/signatures in parallel.

Also if some transactions are processed successfully but others fail, how to we want to represent this partial success/failure? Do we send back an array of results with an entry for each transaction/signature?

Reply:
As receiveTransaction**s** and receiveSignature**s** is used only in P2P layer, we don't care about the response. The error that might occur is only in the transport layer. It is currently implemented as RPC query, but there are no obstacles to changing these (together with blocks) for emit (broadcast) actions, without expecting any response. 

Yes, I think it should be addressed as a separate issue. I will open one.